### ISSUE TITLE
Plugin packaging

### ISSUE THREAD

**Issue by rominf** [oh-my-fish/oh-my-fish#57]
I propose to reuse `git`'s infrastructure as much as possible:

```
$ omf info <plugin>
Name: <plugin name (echo repository_name | sed "s/plugin-//g")>
Version: <git tag; I think one number should be enough for a plugin version>
Installed: Yes
Status: up-to-date <is the tag last?>
Summary: <.git/description content>
Authors: <git commit authors (see http://stackoverflow.com/a/6482473/2108548)>
```

What do you think?

**Comment by rominf** [comment]
Of course, README.md should contain only description. I'm against copy-pasing text like:
```
to install write:
...
to uninstall write:
...
```

I think that:
```
For more information on `oh-my-fish` visit ...
```
would be enough. This text is easy to append automatically by script to all plugins.

**Comment by derekstavis** [comment]
I prefer using git tags instead of branches, this way plugins can be versioned using semver.

**Comment by derekstavis** [comment]
But I definitely like `omf info` command and your proposed output!!

**Comment by rominf** [comment]
> I'd prefer using git tags instead of branches, this way plugins can be versioned using semver.
You are right. I meant tags, not branches. I'll fix it.

**Comment by bpinto** [comment]
I believe I didn't understand the proposal for the README file. Could you try explaining again?

**Comment by rominf** [comment]
@bpinto 
Well. Printing markdown to a console output is not a very good idea.

I actually vote for self-documented code, so I think that after implementation of https://github.com/fish-shell/fish-shell/issues/478 we can just simply build README files from the help. So, for now, I would leave README files as they are.

BTW, it's possible to extract repository description from `.git/description` file.

**Comment by bpinto** [comment]
I can see this working as it was proposed, might be interesting to see what current plugins would output.

**Comment by jeremiejig** [comment]
See also `git describe` to generate a version, which relies on annotated tag.

**Comment by derekstavis** [comment]
I usually use `git describe --tags --always HEAD`

**Comment by bpinto** [comment]
@rominf are you working on a PR for this?

**Comment by rominf** [comment]
@bpinto, no.

**Comment by sagebind** [comment]
This will be marked as solved as part of the effort in #427.

**Comment by sagebind** [comment]
#427 is merged. Packages can now be grouped, be more discoverable, and have metadata using the new package repository system built on top of Git. Closing as solved.
